Data Feed UI consistency updates for list and details page

vinay033 opened the pull request:
### :nut_and_bolt: Description: What code changed, and why?
The new data feed pages updated to be consistent with the rest of Automate.

- [x] Data feed list page
- [x] Data feed create model popup
- [x] Data feed test on model popup
- [x] Data feed delete confirmation
- [x] Data feed details page
- [x] Data feed update on details page 
- [x] Data feed test on details page

### :chains: Related Resources
https://github.com/chef/automate/issues/2098
https://github.com/chef/automate/issues/2746
https://github.com/chef/automate/issues/2760
### :+1: Definition of Done
Added new components for the data feed the same as automate standards and removed unused components. 
### :athletic_shoe: How to Build and Test the Change
```
STEP 1
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

npm run serve:hab
```
This is a hidden feature so you have to enable it first, by turning on the service now feature flag
go to settings tab and where you can see the link on right nav --> `data feed` 
### :white_check_mark: Checklist

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [ ] Docs added/updated?
- [ ] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots
1. List page

![feed-index](https://user-images.githubusercontent.com/12297653/80100056-0b054880-858d-11ea-9312-88f496c07672.png)
2. created a model popup

![Feed-create-popup](https://user-images.githubusercontent.com/12297653/80100135-22443600-858d-11ea-92cb-a3be382d8bfc.png)

3. send test on a model popup with success and failure

![model-feed-tes-failed](https://user-images.githubusercontent.com/12297653/80100163-2ff9bb80-858d-11ea-9d02-e8ca568bbf56.png)
![model-feed-test-pass](https://user-images.githubusercontent.com/12297653/80100168-31c37f00-858d-11ea-876b-729d563532e8.png)

4. create feed notification

![feedCreateNotify](https://user-images.githubusercontent.com/12297653/80100191-3daf4100-858d-11ea-8b22-bec68ad5b236.png)

5. delete feed confirmation

![delete-confirmation](https://user-images.githubusercontent.com/12297653/80100215-4a339980-858d-11ea-8ae1-ecfb048f40a1.png)

6. delete feed notification

![deletefeednotify](https://user-images.githubusercontent.com/12297653/80100226-50c21100-858d-11ea-9bae-6549c2245892.png)
7. Control menus

![buttons](https://user-images.githubusercontent.com/12297653/80100255-591a4c00-858d-11ea-97b1-31c4ba91bd32.png)
8. test feed on list page using the control menu (notification for success and failure)
![testfeedonlistfailure](https://user-images.githubusercontent.com/12297653/80100878-3b011b80-858e-11ea-96f1-e4dc96bf9be3.png)
![testfeedonlistsuccess](https://user-images.githubusercontent.com/12297653/80100884-3ccadf00-858e-11ea-9486-a6e8c321acfe.png)

9. feed details page 
![feedDetails](https://user-images.githubusercontent.com/12297653/80100372-849d3680-858d-11ea-9cf6-5aa181647ca4.png)
10. feed details update with the notification 

![updateFeednotify](https://user-images.githubusercontent.com/12297653/80100399-8ebf3500-858d-11ea-956b-a116d806896c.png)
11. test feed on details page with success and failure notifications 

![feedDetailsTestFaild](https://user-images.githubusercontent.com/12297653/80100440-9aaaf700-858d-11ea-81cc-f2f7beea61c1.png)
![feedDetailsTestsuccess](https://user-images.githubusercontent.com/12297653/80100447-9c74ba80-858d-11ea-9188-da77935cdf7b.png)

susanev commented on components/automate-ui/src/app/entities/destinations/destination.effects.ts:
```diff
@@ -0,0 +1,196 @@
+    map(({ payload }: TestDestinationFailure) => {
+      return new CreateNotification({
+        type: Type.error,
+        message: `We were unable to connect to your data feed ${payload.name}.`
```
```suggestion
        message: `Unable to connect to data feed ${payload.name}.`
```

susanev commented on components/automate-ui/src/app/pages/create-data-feed-modal/create-data-feed-modal.component.html:
```diff
@@ -0,0 +1,91 @@
+              <span class="label">Name <span aria-hidden="true">*</span></span>
+              <input chefInput formControlName="name" type="text" id="name-input" (keyup)="handleInput($event)" autocomplete="off"/>
+            </label>
+            <span class="i-desc">Don't worry, data feed name can be changed later.</span>
```
```suggestion
            <span class="i-desc">Don't worry, data feed names can be changed later.</span>
```

vinay033 commented:
> @vinay033 here is some feedback, i need to take a closer look but wanted to get you some as fast as i could...
> 
> * [x]  help text below inputs should be `font-size: 12px; font-weight: 300;` take a look at the teams create for an example
> * [x]  something is making the labels themselves on the inputs smaller than they should be, they should match with the ones on team create
> * [x]  delete modal heading should be `Delete Data Feed?`
> * [x]  delete modal primary button text should be `Delete Data Feed`
> * [x]  on the detail page remove the left margin on the test button

I have added the changes and updated the PR, please have a look.
